src/noCommentedOutCodeRule.ts

```diff
@@ -72,6 +75,10 @@ class NoCommentedOutCodeRuleWalker extends ErrorTolerantWalker {
     private textIsTslintFlag(text: string): boolean {
         return text.startsWith('tslint:');
     }
+    
+    private textIsNote(text: string): boolean {
+        return /^(NOTE|TODO|FIXME|BUG|HACK|XXX)[:( ]/.test(text);
```

Comment:
Should make this case insensitive 👍 .

Reply:
Case-insensitive is more likely to clash with object properties like `todo: {},`. Is that okay?

Reply:
OK let's leave it case-sensitive. If there's a feature request to make it case-insensitive or even configurable we can discuss it then 😃 .